### ISSUE TITLE
Random failure in Unit.Evolution.Particles.MonteCarlo.InverseJacobianInertialToFluid

### ISSUE THREAD

**Issue by wthrowe** [sxs-collaboration/spectre#5832]
```
2024-03-06T08:31:55.2539917Z Unit.Evolution.Particles.MonteCarlo.InverseJacobianInertialToFluid
2024-03-06T08:31:55.2540981Z -------------------------------------------------------------------------------
2024-03-06T08:31:55.2542120Z ../../../../../../tests/Unit/Evolution/Particles/MonteCarlo/Test_InverseJacobianInertialToFluid.cpp:24
2024-03-06T08:31:55.2543249Z ...............................................................................
2024-03-06T08:31:55.2543758Z 
2024-03-06T08:31:55.2544117Z ../../../../../../tests/Unit/Framework/TestingFramework.hpp:116: FAILED:
2024-03-06T08:31:55.2544850Z   CHECK( a == appx(b) )
2024-03-06T08:31:55.2545267Z with expansion:
2024-03-06T08:31:55.2545662Z   0.00000000001242029 == Approx( 0.0 )
2024-03-06T08:31:55.2546151Z with messages:
2024-03-06T08:31:55.2546771Z   Seed is: 2828983820 from ../../../../../../tests/Unit/Evolution/Particles/
2024-03-06T08:31:55.2547817Z   MonteCarlo/Test_InverseJacobianInertialToFluid.cpp:26
2024-03-06T08:31:55.2548625Z   ../../../../../../tests/Unit/Evolution/Particles/MonteCarlo/
2024-03-06T08:31:55.2549460Z   Test_InverseJacobianInertialToFluid.cpp:90: dot_product ==
2024-03-06T08:31:55.2550181Z   expected_dot_product
2024-03-06T08:31:55.2551086Z   a := (1.24203e-11,5.32907e-15,-2.15106e-16,-5.55112e-17,-2.22045e-16)
2024-03-06T08:31:55.2551822Z   b := (0,0,0,0,0)
2024-03-06T08:31:55.2552072Z 
2024-03-06T08:31:55.2552607Z 0.001 s: Unit.Evolution.Particles.MonteCarlo.InverseJacobianInertialToFluid
2024-03-06T08:31:55.2553799Z ===============================================================================
2024-03-06T08:31:55.2554464Z test cases:   1 |   0 passed | 1 failed
2024-03-06T08:31:55.2555059Z assertions: 127 | 126 passed | 1 failed
```

**Comment by ffoucart** [comment]
I'll run with that seed and have a look.

**Comment by ffoucart** [comment]
High Lorentz factor combined with a very weird metric with large off-diagonal terms... I'm going to commit a PR with some normalization of the error to avoid these issues.

**Comment by wthrowe** [comment]
Fixed in #5833.
